lib/improver/blending/spatial_weights.py

```diff
@@ -0,0 +1,275 @@
+        result = ('<SpatialWeightsForMissingData: fuzzy_length: {}>'.format(
+                      self.fuzzy_length))
+        return result
+
+    @staticmethod
+    def create_initial_weights_from_mask(cube):
+        """
+        Generate a cube with weights generated from the mask of the input cube.
+        Where the data is masked we set the weight to zero, otherwise the
+        weight is one.
+
+        Args:
+            cube (iris.cube.Cube):
+                A cube containing the data we want to collapse by doing a
+                weighted blend along a given cordinate.
+        Return:
+            masked_weights_cube (iris.cube.Cube):
+                A cube containing an initial set of weights based on the mask
+                on the input cube.
+        Rasies:
+            ValueError : If the input cube does not have a mask.
+        """
+        if np.ma.is_masked(cube.data):
+            weights_data = np.where(cube.data.mask, 0, 1).astype(np.float32)
+        else:
+            message = ("Input cube to SpatialWeightsForMissingData "
+                       "must be masked")
+            raise ValueError(message)
+        masked_weights_cube = cube.copy(data=weights_data)
+        masked_weights_cube.rename("weights")
+        return masked_weights_cube
+
+    def create_fuzzy_spatial_weights(self, masked_weights_cube):
+        """
+        Create fuzzy weights around points in the masked_weights_cube with
+        zero weight.
+
+        Args:
+            masked_weights_cube (iris.cube.Cube):
+                A cube containing an initial set of weights based on the mask
+                on the input cube.
+        Returns:
+            result (iris.cube.Cube):
+                A cube contiaing the fuzzy weights calculated based on the
+                masked_weights_cube. The dimension order may have changed from
+                the input cube as it has been sliced over x and y coordinates.
+        """
+        result = iris.cube.CubeList()
+        x_coord = masked_weights_cube.coord(axis='x').name()
+        y_coord = masked_weights_cube.coord(axis='y').name()
+        # The distance_transform_edt works on N-D cubes, so we want to make
+        # sure we only apply it to x-y slices.
+        for weights in masked_weights_cube.slices([x_coord, y_coord]):
+            if np.all(weights.data == 1.0):
+                # distance_transform_edt doesn't produce what we want if there
+                # are no zeros present.
+                result.append(weights.copy())
+            else:
+                fuzzy_data = distance_transform_edt(weights.data == 1., 1)
+                fuzzy_data = fuzzy_data.astype(np.float32)
+                rescaled_fuzzy_data = rescale(
+                    fuzzy_data, data_range=[0., self.fuzzy_length],
+                    clip=True)
+                result.append(weights.copy(data=rescaled_fuzzy_data))
+        result = result.merge_cube()
```

Comment:
I think this function would be better if it maintained the order of the input dimensions, eg by transposing the coordinate order on the merged cube or slicing over y-x in the expected order.